### ISSUE TITLE
Add a `sno revert` command

### ISSUE THREAD

**Issue by craigds** [koordinates/sno#43]
Without a revert command a la `git revert`, it's very difficult to undo an accidental change. Can we add this?

**Comment by olsen232** [comment]
Read some git docs - its clear what the effect of `git revert` is supposed to be, but not exactly how it is implemented
- Seems to performs a cherry-pick, but in reverse

- when git cherry-pick hits conflicts, it lets you resolve them in the same manner as git merge, and it stores the two current heads in HEAD and CHERRY_PICK_HEAD. When git revert hits conflicts, it lets you resolve them in the same manner as git merge, and it stores the two current heads in HEAD and REVERT_HEAD.

pygit2 has a Repository.cherrypick but doesn't have a Repository.revert, but libgit2 has both git_cherrypick and git_revert functions. Adding the missing revert function probably isn't too hard using the cherrypick as a model.

We could maybe support this a bit quicker if we didn't let the user resolve conflicts - the user could revert a commit only if it doesn't conflict, otherwise they have to do it manually. We could do this by, eg, converting a commit to a patch and then inverting the patch, then applying the patch - if it applies. If it doesn't, simply fail with an error message.

**Comment by olsen232** [comment]
Actually, adding the functionality we need to pygit2 is harder than described above, since the existing pygit2 code calls git_cherrypick, which modifies the working copy - we need to call git_revert_commit, which doesn't modify the working copy, only the index:
```
DIRECTION     GIT WORKDIR       PYGIT2 FUNCTION            GIT FUNCTION
Forward       Required          Repository.cherrypick      git_cherrypick
Forward       Not required      Not implemented            git_cherrypick_commit
Backward      Required          Not implemented            git_revert
Backward      Not required      Not implemented <--        git_revert_commit
```

The last row is the pygit2 functionality we would need to implement - we can look at Repository.cherrypick to see how it works, but you can see that it is not as good a template as I first hoped since its not just the patch direction that needs to change, but also whether the git workdir is required. We don't have a git workdir, so we need the `_commit` variant to work.

**Comment by aadityasinha-dotcom** [comment]
Is this issue available? 
Can I work on this issue?

**Comment by olsen232** [comment]
Yes! Be aware that the first part of this fix is to add needed functionality into pygit2, not Kart itself - perhaps pygit2 wrappers on all three functions above for completeness:
git_cherrypick_commit, git_revert and git_revert_commit
